tests/integration-tests/tests/sqlite_authority_tests.rs

```diff
@@ -961,13 +961,11 @@ async fn test_zone_signing() {
         assert!(
             inner_results
                 .iter()
-                .any(|r| r.record_type() == RecordType::RRSIG
-                    && r.name() == record.name()
-                    && if let RData::DNSSEC(DNSSECRData::SIG(ref rrsig)) = *r.data().unwrap() {
-                        rrsig.type_covered() == record.record_type()
-                    } else {
-                        false
-                    }),
+                .filter(|r| r.record_type() == RecordType::RRSIG)
+                .filter(|r| r.name() == record.name())
+                .filter_map(|r| r.data())
+                .filter_map(|r| RRSIG::try_borrow(r).ok())
```

Comment:
This is pretty ugly. In general, IMO subsequent `.filter().filter()` or `filter().filter_map()` calls are a bit smelly, I usually prefer to put everything in the same closure.

Reply:
fair, I can combine into a single filter_map, I'll see how that looks.

Reply:
Ok, I combined the filter statements, the nice thing about the filter_maps is that they reduce the amount of code for performing the type mapping conversions.

Reply:
I actually meant that we should collapse the whole `.filter().filter().filter_map().filter_map()` contraption into a single `filter_map()`.

Reply:
see recent edits.